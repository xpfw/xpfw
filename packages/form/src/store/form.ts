import { get, set } from "lodash"
import { action, observable } from "mobx"
import { prependPrefix } from "../util/prefixMaker"

/**
 * Holds data relevant for Forms such as values and validation errors
 */
export class FormStoreClass {
  /**
   * MobX Observable containing all form data
   */
  @observable
  public formData: any = {}
  /**
   * MobX Observable containing all form errors
   */
  @observable
  public errors: any = {}
  /**
   * MobX Observable containing Loading states
   */
  @observable
  public loading: any = {}

  /**
   * Acquire a value from the form data
   * @param valuePath key where the value of the field will be saved. uses lodash.get/set string syntax style so subfields, array indices etc. can be used within the string
   * @param prefix prepended to mapTo to allow same mapTo keys to have different values
   * @param defaultValue value to return if it is not set in the form data
   */
  public getValue(valuePath?: string, prefix?: string, defaultValue?: any): any {
    return get(this.formData, prependPrefix(valuePath, prefix), defaultValue)
  }

  /**
   * Set a value in the form data
   * @param valuePath key where the value of the field will be saved. uses lodash.get/set string syntax style so subfields, array indices etc. can be used within the string
   * @param value the new value
   * @param prefix prepended to mapTo to allow same mapTo keys to have different values
   */
  @action
  public setValue(valuePath?: string, value?: any, prefix?: string) {
    set(this.formData, prependPrefix(valuePath, prefix), value)
  }

  /**
   * Acquire a value from the form errors
   * @param valuePath key where the value of the field will be saved. uses lodash.get/set string syntax style so subfields, array indices etc. can be used within the string
   * @param prefix prepended to mapTo to allow same mapTo keys to have different values
   */
  public getError(valuePath?: string, prefix?: string): any {
    return get(this.errors, prependPrefix(valuePath, prefix))
  }

  /**
   * Set a value in the form errors
   * @param valuePath key where the value of the field will be saved. uses lodash.get/set string syntax style so subfields, array indices etc. can be used within the string
   * @param value the new value
   * @param prefix prepended to mapTo to allow same mapTo keys to have different values
   */
  @action
  public setError(valuePath?: string, value?: any, prefix?: string) {
    set(this.errors, prependPrefix(valuePath, prefix), value)
  }

  /**
   * Acquire a value from the form errors
   * @param valuePath key where the value of the field will be saved. uses lodash.get/set string syntax style so subfields, array indices etc. can be used within the string
   * @param prefix prepended to mapTo to allow same mapTo keys to have different values
   */
  public getLoading(valuePath?: string, prefix?: string): any {
    return get(this.loading, prependPrefix(valuePath, prefix))
  }

  /**
   * Set a value in the form errors
   * @param valuePath key where the value of the field will be saved. uses lodash.get/set string syntax style so subfields, array indices etc. can be used within the string
   * @param value the new value
   * @param prefix prepended to mapTo to allow same mapTo keys to have different values
   */
  @action
  public setLoading(valuePath: string, value: boolean, prefix?: string) {
    set(this.loading, prependPrefix(valuePath, prefix), value)
  }
}

const FormStore = new FormStoreClass()
export default FormStore
