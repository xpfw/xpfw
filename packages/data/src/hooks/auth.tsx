import { ExtendedJSONSchema, JSONSchemaDefinition } from "@xpfw/form"
import UserStore from "../store/user"

const MailField: JSONSchemaDefinition = {
  title: "email",
  type: "string"
}
const PwField: JSONSchemaDefinition = {
  title: "password",
  type: "string",
  format: "password"
}

/**
 * Be sure to adjust relationship(Id|NamePath) to your user object
 */
const OwnerField: ExtendedJSONSchema = {
  title: "belongsTo",
  type: "relationshipSingle",
  relationship: {
    namePath: MailField.title,
    collection: "user"
  }
}

const AuthForm: ExtendedJSONSchema = {
  title: "authForm",
  collection: "user",
  required: [String(MailField.title), String(PwField.title)],
  properties: {
    [String(MailField.title)]: MailField,
    [String(PwField.title)]: PwField
  }
}

const submitLogin = () => UserStore.login()
const submitRegister = async () => {
  const registerRes = await UserStore.register()
  await submitLogin()
  return registerRes
}
const submitLogout = () => UserStore.logout()

const useAuth = () => {
  return {
    loading: UserStore.getLoading(),
    user: UserStore.getUser(),
    loggedIn: UserStore.getLoggedIn(),
    error: UserStore.getAuthErr(),
    connected: UserStore.getConnected(),
    submitLogout, submitLogin, submitRegister
  }
}

export default useAuth
export {
  AuthForm, PwField, MailField, OwnerField,
  submitLogin, submitRegister, submitLogout
}
