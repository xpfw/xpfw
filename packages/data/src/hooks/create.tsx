import { ExtendedJSONSchema, FormStore, getMapTo, jsonValidator, memo, prependPrefix } from "@xpfw/form"
import { get, isNil } from "lodash-es"
import { useEffect } from "react"
import DbStore from "../store/db"
import UserStore from "../store/user"

const submitCreate = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  return async () => {
    if (mapTo == null) {
      mapTo = getMapTo(schema, mapTo)
    }
    const newValue = FormStore.getValue(mapTo, prefix)
    jsonValidator.validate(schema, newValue)
    FormStore.setError(mapTo, jsonValidator.errors, prefix)
    const err = FormStore.getError(mapTo, prefix)
    let res
    if (err != null) {
      res = await DbStore.create(schema, prefix)
    }
    return res
  }
}
const u: any = DbStore
const useCreate = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, reset?: boolean) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  useEffect(() => {
    if (reset) {
      u.createState[prependPrefix(mapTo, prefix)] = null
      FormStore.setValue(mapTo, {}, prefix)
      FormStore.setError(mapTo, undefined, prefix)
    }
  }, [reset])
  return {
    error: FormStore.getError(mapTo, prefix),
    state: DbStore.getCreateState(mapTo, prefix),
    loading: FormStore.getLoading(mapTo, prefix),
    user: UserStore.getUser(),
    submitCreate: memo(() => submitCreate(schema, mapTo, prefix), [JSON.stringify(schema), mapTo, prefix])
  }
}

export default useCreate
export {
  submitCreate
}
