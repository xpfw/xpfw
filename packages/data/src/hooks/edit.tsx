import { ExtendedJSONSchema, FormStore, getMapTo, jsonValidator, memo, prependPrefix } from "@xpfw/form"
import { useEffect } from "react"
import DbStore from "../store/db"
import UserStore from "../store/user"
import toJS from "../util/toJS"

const submitEdit = (id: string, schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  return () => {
    if (mapTo == null) {
      mapTo = getMapTo(schema, mapTo)
    }
    const newValue = toJS(FormStore.getValue(mapTo, prefix))
    jsonValidator.validate(schema, newValue)
    FormStore.setError(mapTo + id, jsonValidator.errors, prefix)
    const err = FormStore.getError(mapTo + id, prefix)
    if (err == null) {
      return DbStore.patch(id, schema, mapTo, prefix)
    }
    return err
  }
}
const u: any = DbStore
const useEdit = (id: string, schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, reset?: boolean) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  const valuePath = `${prependPrefix(mapTo, prefix)}${id}`
  useEffect(() => {
    if (reset) {
      u.currentlyEditing = ""
      u.updateState[prependPrefix(mapTo, prefix)] = null
      FormStore.setValue(mapTo, {}, prefix)
    }
  }, [reset])
  return {
    error: FormStore.getError(valuePath),
    state: DbStore.getUpdateState(valuePath),
    loading: FormStore.getLoading(valuePath),
    original: DbStore.getEditOriginal(id, schema, mapTo, prefix, false),
    user: UserStore.getUser(),
    submitEdit: memo(() => submitEdit(id, schema, mapTo, prefix),
      ["submitEdit", id, JSON.stringify(schema), mapTo, prefix])
  }
}

export interface IEditHookProps {
  schema: ExtendedJSONSchema
  id: string
  prefix?: string
  mapTo?: string
  reset?: boolean
}

const useEditWithProps = (props: IEditHookProps) => useEdit(props.id, props.schema, props.mapTo, props.prefix, props.reset)

export default useEdit
export {
  useEditWithProps, submitEdit
}
