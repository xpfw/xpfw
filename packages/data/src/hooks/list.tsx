import { ExtendedJSONSchema, FormStore, getMapTo, prependPrefix } from "@xpfw/form"
import { isNil } from "lodash"
import DbStore from "../store/db"
import ListStore from "../store/list"

const nextPage = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  return () => {
    return ListStore.nextPage(schema, mapTo, prefix)
  }
}

const prevPage = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  return () => {
    return ListStore.prevPage(schema, mapTo, prefix)
  }
}

export interface IListOptions {
  defaultEntries?: any[]
}

export interface IListHookProps {
  schema: ExtendedJSONSchema
  mapTo?: string
  prefix?: string
  options?: IListOptions
}

const useList = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, options?: IListOptions) => {
  const queryObj = ListStore.buildQueryObj(schema, mapTo, prefix, true)
  if (mapTo == null) {
    mapTo = getMapTo(schema, mapTo)
  }
  const getKey = prependPrefix(mapTo, prefix)
  const loading = FormStore.getLoading(getKey)
  const currentPage = ListStore.getCurrentPage(getKey)
  const maxPage = ListStore.getMaxPage(getKey)
  // needed so componentDidUpdate gets triggered on form value change
  let list: any = ListStore.getList(schema, mapTo, prefix)
  if (isNil(list) || Array.isArray(list.data) && list.data.length === 0)  {
    if (options != null && Array.isArray(options.defaultEntries)) {
      const data = []
      for (const entry of options.defaultEntries) {
        const fetchRes = DbStore.getGetState(entry, String(schema.collection), true)
        if (!isNil(fetchRes)) {
          data.push(fetchRes)
        }
      }
      list = {data, total: 0, isDefaultList: true}
    }
  }
  return {
    list,
    loading,
    currentPage, maxPage,
    queryData: queryObj,
    showNextPage: currentPage < maxPage - 1,
    showPrevPage: currentPage > 0
  }
}

const useListWithProps = (props: IListHookProps) => useList(props.schema, props.mapTo, props.prefix, props.options)

export default useList
export {
  nextPage, prevPage, useListWithProps
}
