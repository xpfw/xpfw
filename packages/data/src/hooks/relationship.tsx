import { ExtendedJSONSchema, FormStore, getMapTo, IFieldProps, memo, prependPrefix, useField, useFieldWithValidation } from "@xpfw/form"
import { cloneDeep, get, isArray, isNil, isNumber, isString, remove } from "lodash"
import { action } from "mobx"
import * as React from "react"
import dataOptions from "../options"
import DbStore from "../store/db"
import ListStore from "../store/list"

const lastUsedKey = "lastUsed"

const getListFormFromRelationshipField:
(schema: ExtendedJSONSchema, mapTo?: string) => ExtendedJSONSchema = (schema, mapTo) => {
  const collection = get(schema, "relationship.collection")
  const nameTransform: any = get(schema, "relationship.nameTransform", (schema: any, val: any) => {
    if (val == null || val.length === 0) {
      return null
    }
    return {
      $regex: `(.*?)${val}(.*?)`,
      $options: "isg"
    }
  })
  const nameSearchField: ExtendedJSONSchema = {
    title: get(schema, "relationship.namePath"),
    type: "string"
  }
  return {
    title: "searchFor." + getMapTo(schema, mapTo),
    collection,
    properties: {
      [String(nameSearchField.title)]: nameSearchField,
      [dataOptions.idPath]: {
        title: dataOptions.idPath,
        type: "string"
      }
    },
    relationship: {
      nameTransform
    }
  }
}

const displayModeChanger = (mapTo: string, prefix?: string) => {
  return (newValue: any) => {
    FormStore.setValue(`displayMode.${mapTo}`, newValue, prefix)
  }
}

const addId = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  const disp = displayModeChanger(mapTo, prefix)
  return action((newValue: any) => {
    if (schema.type === "array") {
      let currentValue = FormStore.getValue(mapTo, prefix)
      currentValue = !isArray(currentValue) ? [] : currentValue
      currentValue.push(newValue)
      FormStore.setValue(mapTo, currentValue, prefix)
      disp(0)
    } else {
      FormStore.setValue(mapTo, newValue, prefix)
    }
    const lastUsedFieldKey = `${lastUsedKey}.${prependPrefix(mapTo, prefix)}`
    let lastUsed = FormStore.getValue(lastUsedFieldKey)
    if (!Array.isArray(lastUsed)) {
      lastUsed = []
    }
    const indexOfLastUsed = lastUsed.indexOf(newValue)
    if (indexOfLastUsed !== -1) {
      lastUsed.splice(indexOfLastUsed, 1)
    }
    lastUsed.unshift(newValue)
    if (lastUsed.length > ListStore.pageSize) {
      lastUsed.splice(1)
    }
    FormStore.setValue(lastUsedFieldKey, lastUsed)
  })
}

const removeId = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  return action((newValue: any) => {
    if (schema.type === "array") {
      let currentValue = FormStore.getValue(mapTo)
      currentValue = !Array.isArray(currentValue) ? [] : cloneDeep(currentValue)
      remove(currentValue, (val) => val === newValue)
      FormStore.setValue(mapTo, currentValue, prefix)
    } else {
      FormStore.setValue(mapTo, undefined, prefix)
    }
  })
}

const searchRelated = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  return async (newValue: any) => {
    const form = getListFormFromRelationshipField(schema)
    const value = FormStore.getValue(mapTo, prefix)
    if (get(schema.relationship, "filterOutSelected", false) === true) {
      const f = useField(dataOptions.idPath, prependPrefix(mapTo, prefix))
      f.setValue({$nin: Array.isArray(value) ? value : [value]})
    }
    const nameField = useField(get(schema, "relationship.namePath"), prependPrefix(form.title, prefix))
    const transformer = get(form, "relationship.nameTransform")
    nameField.setValue(transformer(schema, newValue))
    const res: any = await ListStore.getList(form, undefined, prefix, true)
    if (get(schema.relationship, "autoSelect", false) === true) {
      if (!isNil(res) && Array.isArray(res.data) && res.data.length === 1) {
        addId(schema, mapTo, prefix)(get(res.data[0], dataOptions.idPath))
      }
    }
    return res
  }
}

// public componentDidUpdate() {
//   const prefix = this.props.prefix
//   const field: any = get(this, "props.field")
//   const form = getListFormFromRelationshipField(field)
//   const preppedPrefix = prefix && prefix.length > 0 ? prefix + "." : ""
//   const value = get(this, "props.value", "")
//   if (value.length > 0 && !isNil(form.sections[0].fields[1])) {
//     FormStore.setValue(preppedPrefix + field.mapTo + "." + form.sections[0].fields[1].mapTo,
//       {$nin: Array.isArray(value) ? value : [value]})
//   }
// }

const useRelationship = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  const hookedField = useFieldWithValidation(schema, mapTo, prefix)
  const value: any = hookedField.value
  const collection: any = get(schema.relationship, "collection", "")
  // TODO: use prefix
  const displayMode = FormStore.getValue(`displayMode.${mapTo}`)
  let relatedObject: any
  const searchForm = getListFormFromRelationshipField(schema)
  if (isArray(value)) {
    relatedObject = []
    for (const id of value) {
      if (!isNil(id)) {
        relatedObject.push(DbStore.get(id, collection, false))
      }
    }
  } else if (isNumber(value) || isString(value) && value.length > 0) {
    const arg: any = value
    relatedObject = DbStore.get(arg, collection, false)
  }
  const lastUsed = FormStore.getValue(`${lastUsedKey}.${prependPrefix(mapTo, prefix)}`)
  // becausae autoSelect is the only relevant property in schema we only check against this instead of the stringifed schema for speed
  const autoSelect = get(schema.relationship, "autoSelect")
  return {
    value, setValue: hookedField.setValue, relatedObject, searchForm, displayMode, lastUsed,
    addId: memo(() => addId(schema, mapTo, prefix), ["addId", mapTo, prefix, autoSelect]), prefix,
    removeId: memo(() => removeId(schema, mapTo, prefix), ["removeId", mapTo, prefix, autoSelect]),
    searchRelated: memo(() => searchRelated(schema, mapTo, prefix), ["searchRelated", mapTo, prefix, autoSelect]),
    showDisplay: memo(() => displayModeChanger(String(mapTo), prefix)(1), ["showDisplay", mapTo, prefix]),
    hideDisplay: memo(() => displayModeChanger(String(mapTo), prefix)(0), ["hideDisplay", mapTo, prefix])
  }
}

export interface IRelationshipHookProps {
  schema: ExtendedJSONSchema
  mapTo?: string
  prefix?: string
}

const useRelationshipWithProps = (props: IRelationshipHookProps) => useRelationship(props.schema, props.mapTo, props.prefix)

export default useRelationship
export {
  addId, removeId, searchRelated, getListFormFromRelationshipField, useRelationshipWithProps, displayModeChanger
}
