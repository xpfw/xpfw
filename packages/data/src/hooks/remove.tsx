import { ExtendedJSONSchema, FormStore, memo } from "@xpfw/form"
import DbStore, { REMOVE_ADDON_KEY } from "../store/db"

const submitRemove = (id: string, schema: ExtendedJSONSchema) => {
  return async () => {
    return DbStore.remove(id, String(schema.collection))
  }
}

const useRemove = (id: string, schema: ExtendedJSONSchema) => {
  return {
    state: DbStore.getRemoveState(id),
    loading: FormStore.getLoading(REMOVE_ADDON_KEY + id),
    submitRemove: memo(() => submitRemove(id, schema), [id, String(schema.collection)])
  }
}

export interface IRemoveHookProps {
  schema: ExtendedJSONSchema
  id: string
}

const useRemoveWithProps = (props: IRemoveHookProps) => useRemove(props.id, props.schema)

export default useRemove
export {
  useRemoveWithProps, submitRemove
}
