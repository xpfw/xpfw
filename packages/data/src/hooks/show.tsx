import { FormStore } from "@xpfw/form"
import DbStore from "../store/db"

const useGet = (id: string, collection: string) => {
  const loading = FormStore.getLoading(id)
  return {
    item: DbStore.getGetState(id, collection, true),
    loading
  }
}

export default useGet
