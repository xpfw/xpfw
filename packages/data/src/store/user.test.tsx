import { FeathersClient } from "@xpfw/data-feathers"
import { toJS } from "@xpfw/data-tests"
import { FormStore, useFieldWithValidation, useObject } from "@xpfw/form"
import { getRandomApp } from "@xpfw/test-util"
import "isomorphic-fetch"
import BackendClient from "../client"
import { AuthForm, MailField, PwField } from "../hooks/auth"
import UserStore from "./user"

BackendClient.client = FeathersClient

test("User Store Test", async () => {
  const appRef = await getRandomApp("a", false, BackendClient.client, true)
  const authHook = useObject(AuthForm)
  const mailHook = useFieldWithValidation(authHook.fields[0].schema)
  const pwHook = useFieldWithValidation(authHook.fields[1].schema)
  mailHook.setValue("admin")
  pwHook.setValue("admin")
  expect(toJS(UserStore)).toMatchSnapshot("before register")
  expect(toJS(FormStore.formData)).toMatchSnapshot("before registerform")
  const registerRes = await UserStore.register()
  expect(registerRes).toMatchSnapshot("result of registration")
  expect(toJS(FormStore.formData)).toMatchSnapshot("after register form")
  expect(toJS(UserStore)).toMatchSnapshot("after register")
  await UserStore.login()
  expect(toJS(UserStore)).toMatchSnapshot("after login")
  expect(toJS(FormStore.formData)).toMatchSnapshot("after login form")
  await UserStore.logout()
  expect(toJS(UserStore)).toMatchSnapshot("after logout")
  pwHook.setValue("admin")
  await UserStore.login()
  expect(toJS(UserStore)).toMatchSnapshot("after relogin")
  expect(toJS(FormStore.formData)).toMatchSnapshot("after relogin form")
  await UserStore.logout()
  expect(toJS(UserStore)).toMatchSnapshot("after logout")
  const invalidLogin = await UserStore.login()
  expect(UserStore.getLoading()).toBeFalsy()
  expect(UserStore.getLoggedIn()).toBeFalsy()
  expect(UserStore.getUser()).toBeNull()
  expect(UserStore.getAuthErr() == null)
  const client = BackendClient.client.client
  BackendClient.client.client = null
  const reRegister = await UserStore.register()
  expect(toJS(UserStore)).toMatchSnapshot("after register err")
  BackendClient.client.client = client

  await expect(BackendClient.client.connectTo(`http://localhost:${appRef.port}`, {
    makeAuth: true, userStore: UserStore, useRest: true
  })).rejects.toMatchSnapshot("makeauth")

  await appRef.cleanUp()
}, 10000)
