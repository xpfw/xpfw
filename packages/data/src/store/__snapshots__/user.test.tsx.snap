// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Store Test: after login 1`] = `
UserStore {
  "authErr": null,
  "connected": false,
  "loading": false,
  "loggedIn": true,
  "user": Object {
    "email": "admin",
    "id": 0,
  },
}
`;

exports[`User Store Test: after login form 1`] = `
Object {
  "authForm": Object {
    "email": "admin",
    "password": "",
    "strategy": "local",
  },
}
`;

exports[`User Store Test: after logout 1`] = `
UserStore {
  "authErr": null,
  "connected": false,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`User Store Test: after logout 2`] = `
UserStore {
  "authErr": null,
  "connected": false,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`User Store Test: after register 1`] = `
UserStore {
  "authErr": null,
  "connected": false,
  "loading": false,
  "loggedIn": true,
  "user": Object {
    "email": "admin",
    "id": 0,
  },
}
`;

exports[`User Store Test: after register err 1`] = `
UserStore {
  "authErr": "TypeError: Cannot read property 'service' of null",
  "connected": false,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`User Store Test: after register form 1`] = `
Object {
  "authForm": Object {
    "email": "admin",
    "password": "admin",
  },
}
`;

exports[`User Store Test: after relogin 1`] = `
UserStore {
  "authErr": null,
  "connected": false,
  "loading": false,
  "loggedIn": true,
  "user": Object {
    "email": "admin",
    "id": 0,
  },
}
`;

exports[`User Store Test: after relogin form 1`] = `
Object {
  "authForm": Object {
    "email": "admin",
    "password": "",
    "strategy": "local",
  },
}
`;

exports[`User Store Test: before register 1`] = `
UserStore {
  "authErr": null,
  "connected": false,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`User Store Test: before registerform 1`] = `
Object {
  "authForm": Object {
    "email": "admin",
    "password": "admin",
  },
}
`;

exports[`User Store Test: makeauth 1`] = `[NotAuthenticated: No accessToken found in storage]`;

exports[`User Store Test: result of registration 1`] = `
Object {
  "email": "admin",
  "id": 0,
}
`;
